packages/angular/src/generators/scam/scam.ts

```diff
@@ -21,4 +29,20 @@ export async function scamGenerator(tree: Tree, schema: Schema) {
   await formatFiles(tree);
 }
 
+function checkPathUnderProjectRoot(tree: Tree, options: Partial<Schema>) {
+  const project =
+    options.project ?? readWorkspaceConfiguration(tree).defaultProject;
+  const { sourceRoot } = readProjectConfiguration(tree, project);
+
+  if (!options.path) {
```

Comment:
Not a big deal, but it's probably better to move this to the top to bailout earlier and avoid reading configuration if not needed.

Reply:
good idea!